### PR TITLE
Shot: Add ability to edit in PixelPaint

### PR DESCRIPTION
**Pull request by borgmanJeremy**
Adding the ability to directly edit a screenshot via PixelPaint. This seemed like a good way to allow editing a screenshot without significantly complicating the shot code.

### CODE REVIEW

File: Userland/Utilities/shot.cpp [hunk 1]
```diff
@@ -152,6 +154,9 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         warnln("Failed to encode PNG");
         return 1;
     }
+    if (edit_image) {
+        output_path = Core::DateTime::now().to_string("/tmp/screenshot-%Y-%m-%d-%H-%M-%S.png"sv);
+    }
```
Comment (MacDue) [inline Userland/Utilities/shot.cpp:159]:
```suggestion
    if (edit_image)
        output_path = Core::DateTime::now().to_string("/tmp/screenshot-%Y-%m-%d-%H-%M-%S.png"sv);
```
No braces necessary.

File: Userland/Utilities/shot.cpp [hunk 2]
```diff
@@ -165,6 +170,13 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         return 1;
     }
 
+    if (edit_image) {
+        char const* argv[] = { "PixelPaint", output_path.characters() };
+        pid_t child_pid;
+        if ((errno = posix_spawn(&child_pid, "/bin/PixelPaint", nullptr, nullptr, const_cast<char**>(argv), environ)))
+            perror("posix_spawn");
```
Comment (MacDue) [inline Userland/Utilities/shot.cpp:177]: This is one line with the wrappers in LibCore :^)
```suggestion
        TRY(Core::Process::spawn("/bin/PixelPaint", Array { output_path });
```

**Review by MacDue — CHANGES_REQUESTED** [review]

File: Userland/Utilities/shot.cpp (hunk 2 above)
Comment (MacDue) [reply Userland/Utilities/shot.cpp:177]: (You will need to `#include <LibCore/Process.h>`)

Comment (MacDue) [reply Userland/Utilities/shot.cpp:177]: Also, I just spotted your `argv` is non-nullptr terminated which could lead to a bad time... Another reason why you should prefer the wrappers.

**Review by networkException — CHANGES_REQUESTED** [review]
Please use lowercase "shot: " to keep this consistent with other utilities commits

File: Userland/Utilities/shot.cpp (hunk 2 above)
Comment (borgmanJeremy) [reply Userland/Utilities/shot.cpp]: Thanks for the feedback! I'm, new to the codebase so I was not aware of that macro.

Changes have been addressed.

File: Userland/Utilities/shot.cpp [hunk 3]
```diff
@@ -165,25 +170,30 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         return 1;
     }
 
+    if (edit_image)
+        TRY(Core::Process::spawn("/bin/PixelPaint", Array { output_path })
```
[the remark this answers is not in the document]
Comment (MacDue) [reply Userland/Utilities/shot.cpp:174]: I believe you'll also have to add the `sv` suffix to the path

[the remark this answers is not in the document]
Comment (borgmanJeremy) [reply Userland/Utilities/shot.cpp]: Sorry for wasting time, I was in a hurry and should have waited until I had more time.

[the remark this answers is not in the document]
Comment (MacDue) [reply Userland/Utilities/shot.cpp]: No worries, feel free to press the "resolved" button on my comments now :^)

File: Base/usr/share/man/man1/shot.md [hunk 4]
```diff
@@ -14,6 +14,7 @@ $ shot [--clipboard] [--delay seconds] [--screen index] [--region] [output]
 * `-d seconds`, `--delay seconds`: Seconds to wait before taking a screenshot
 * `-s index`, `--screen index`: The index of the screen (default: -1 for all screens)
 * `-r`, `--region`: Select a region to capture
+* `-e`  `--edit`: Open in PixelPaint
```
Comment (networkException) [inline Base/usr/share/man/man1/shot.md:17]:
```suggestion
* `-e`, `--edit`: Open in PixelPaint
```

**Review by networkException — APPROVED** [review]
very nice!

**Pull request merged**